apis/s3/v1alpha2/bucketpolicy_types.go

```diff
@@ -48,147 +47,133 @@ type BucketPolicyParameters struct {
 	// BucketNameSelector selects a reference to an S3Bucket to retrieve its bucketName
 	// +optional
 	BucketNameSelector *runtimev1alpha1.Selector `json:"bucketNameSelector,omitempty"`
-
-	// UserName presents the name of the IAM user this BucketPolicy is concerned with.
-	// +optional
-	UserName *string `json:"userName,omitempty"`
-
-	// UserNameRef references to an S3Bucket to retrieve its userName
-	// +optional
-	UserNameRef *runtimev1alpha1.Reference `json:"userNameRef,omitempty"`
-
-	// UserNameSelector selects a reference to an S3Bucket to retrieve its userName
-	// +optional
-	UserNameSelector *runtimev1alpha1.Selector `json:"userNameSelector,omitempty"`
-}
-
-// Serialize is the custom marshaller for the BucketPolicyParameters
-func (p *BucketPolicyParameters) Serialize() (interface{}, error) {
-	m := make(map[string]interface{})
-	m["Version"] = p.PolicyVersion
-	if p.PolicyID != "" {
-		m["Id"] = p.PolicyID
-	}
-	slc := make([]interface{}, len(p.PolicyStatement))
-	for i, v := range p.PolicyStatement {
-		msg, err := v.Serialize()
-		if err != nil {
-			return nil, err
-		}
-		slc[i] = msg
-	}
-	m["Statement"] = slc
-	return m, nil
 }
 
 // BucketPolicyStatement defines an individual statement within the
 // BucketPolicyBody
 type BucketPolicyStatement struct {
 	// Optional identifier for this statement, must be unique within the
 	// policy if provided.
-	StatementID string `json:"sid,omitempty"`
+	// +optional
+	StatementID *string `json:"sid,omitempty"`
```

Comment:
```suggestion
	StatementID *string `json:"statementId,omitempty"`
```

Is there a specific reason you want the jsontag be different than Go field name?

Reply:
It was more for clarity, in the actual S3 bucket policy json, the field is "sid" and not "statementId", but I can switch it.

Reply:
Gotcha. Then it seems like changing Go field names makes more sense since we'd like to keep the UX of IAM policy as is. Do all the jsontags map to classic policy fields right now after the last push?

Reply:
Yeah, that is what I ended up doing, I think this makes it more clear from the developer/maintainer experience as well. These changes have been pushed :)